### PR TITLE
Implement counter metric type for Python bindings.

### PR DESCRIPTION
**Pull request by mdboom**
It's pretty helpful to have type annotations for the documentation, if nothing
else, so additionally, this adds support for mypy type annotations.

### CODE REVIEW

File: glean-core/android/src/test/java/mozilla/telemetry/glean/private/CounterMetricTypeTest.kt [hunk 1]
```diff
@@ -56,25 +56,6 @@ class CounterMetricTypeTest {
...
-            name = "counter_metric",
-            sendInPings = listOf("store1")
-        )
-
-        // Attempt to increment the counter
-        counterMetric.add(1)
-        // Check that nothing was recorded.
-        assertFalse("Counters must not be recorded if they are disabled",
-            counterMetric.testHasValue())
-    }
```
Comment (mdboom) [inline glean-core/android/src/test/java/mozilla/telemetry/glean/private/CounterMetricTypeTest.kt]: This test is *exactly* the same as the one below it (except the name and the comment), so I just went ahead and removed it.  Save a few cycles per test ;)

File: .circleci/config.yml [hunk 2]
```diff
@@ -492,14 +492,14 @@ jobs:
             npm install -g --silent gh-pages@2.0.1
             gh-pages --dotfiles --message "[skip ci] Updates" --dist build/docs
 
-  Lint Python with flake8 and black:
+  Lint Python with flake8 and black and mypy:
```
Comment (badboy) [inline .circleci/config.yml]: This becomes unhandy. Should we just name it `Lint Python` and not bother to change it with every new tool we add?

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+max-line-length = 88
```
Comment (badboy) [inline .flake8]: :/
Any reason we don't go straight to 100?

File: glean-core/python/glean/_ffi.py [hunk 4]
```diff
@@ -64,4 +66,25 @@ def make_config(data_dir, package_name):
...
+    values.append(ffi.NULL)
+
+    result = ffi.new("char *[]", values)
+
+    _global_weakkeydict[result] = values
+
+    return result
+
+
+__all__ = ["ffi", "ffi_string", "ffi_vec_string", "lib", "make_config"]
```
Comment (badboy) [inline glean-core/python/glean/_ffi.py]: What does `__all__` do? Just mark what functions are exported?

File: .flake8 (hunk 3 above)
Comment (Dexterp37) [inline .flake8]: How was 88 picked? Does this come from a PEP or something?

FWIW Mozilla has [these guidelines](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style#Python_Practices) for Python.

File: glean-core/python/glean/metrics/counter.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        Add to counter value.
+
+        Args:
+            amount (int): (default: 1) This is the amount to increment the
+                counter by.
+        """
+        if self._disabled:
+            return
+
+        # TODO: Run on the dispatcher
```
Comment (Dexterp37) [inline glean-core/python/glean/metrics/counter.py]: nit: can we reference the bug number, here?

**Pull request merged**